modules/adagioAnalyticsAdapter.js

```diff
@@ -129,6 +130,31 @@ function addKeyPrefix(obj, prefix) {
   }, {});
 }
 
+function getCurrencyData(bid) {
+  const result = {
+    netCpm: bid.cpm,
+    orginalCpm: bid.originalCpm
+  }
+
+  if (typeof bid.currency === 'string' && bid.currency.toUpperCase() !== CURRENCY_USD) {
+    if (typeof getGlobal().convertCurrency === 'function') {
+      result.netCpm = parseFloat(Number(getGlobal().convertCurrency(bid.cpm, bid.currency, CURRENCY_USD))).toFixed(3);
+    } else {
+      result.netCpm = null
+    }
+  }
+
+  if (typeof bid.originalCurrency === 'string' && bid.originalCurrency.toUpperCase() !== CURRENCY_USD) {
+    if (typeof getGlobal().convertCurrency === 'function') {
+      result.orginalCpm = parseFloat(Number(getGlobal().convertCurrency(bid.originalCpm, bid.originalCurrency, CURRENCY_USD))).toFixed(3);
+    } else {
+      result.orginalCpm = null
+    }
+  }
+
+  return result
+}
+
```

Comment:
Can we refactor? the two blocs inside the func are really the same

Reply:
Yes you can

Reply:
I'll let you make the modification